pyccel/ast/variable.py

```diff
@@ -252,7 +239,7 @@ def shape_can_change(self, i):
         """
         Indicates if the shape can change in the i-th dimension
         """
-        return self.is_pointer
+        return self.is_alias
```

Comment:
I don't think so. As you say in the future we could use an aliased variable to reference a mutable scalar object which is not an array so the alias concept is not specific to arrays.

We could add an `is_aliased_array` if you think this is useful for something? But I'm not sure what the use case is

Reply:
No need to do this change?

Reply:
I think that this property is specific to arrays. Please let me know if I am wrong.
This is the whole code:
```python
    def shape_can_change(self, i):
        """
        Indicates if the shape can change in the i-th dimension
        """
        return self.is_alias
```

Reply:
I like `self.rank > i`!
BTW, are we willing to pay a small price and assert that `i >= 0`?

Reply:
This is ok with me, but I am not sure it is necessary. The only place it is used is [here](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/variable.py#L237) unless that changes it will always pass these assertions anyway. That said, looking at the code there, I think it ought to be `return self.is_stack_array`. A heap array can also change size if it is reallocated

Reply:
The whole `process_shape` function looks a bit odd. But I think that's outside the scope of this PR.
Personally I would prefer leaving this function as is as that makes sense for this PR and then separately (probably in #1056 where this code has already caused problems) change `process_shape` to not need this function and then remove `shape_can_change`

Reply:
Fine with me! There is no need to change this code then.

Reply:
it should be like this? 
```python
return self.is_alias and self.rank > i
```
or just check for `rank`?

Reply:
@aihya Unfortunately GitHub does not update the page automatically, and one needs to refresh it manually. So you probably have not seen my last message.

There is no need to change the code in this PR. I am closing this discussion now.